sherpa/sim/__init__.py

```diff
@@ -758,3 +761,89 @@ def calc_stat(proposed_params):
         stats = -2.0 * stats
 
         return (stats, accept, params)
+
+
+class ReSampleData(NoNewAttributesAfterInit):
+
+    def __init__(self, data, model):
+        self.data = data
+        self.model = model
+        NoNewAttributesAfterInit.__init__(self)
+        return
+    
+    def __call__(self, niter=1000, seed=123):
```

Comment:
I think it would be better to have `seed=None` rather than `seed=123`, since it will then use a "random" seed by default. Or is that not the intended use case?

Reply:
will change the seed from 123 to None